src/v/storage/log_manager.cc

```diff
@@ -358,7 +365,7 @@ std::ostream& operator<<(std::ostream& o, const log_config& c) {
         o << "nullopt";
     }
     return o << ", compaction_interval_ms:" << c.compaction_interval().count()
-             << ", delete_reteion_ms:" << c.delete_retention().count()
+             << ", delete_reteion_ms:" << c.delete_retention()->count()
```

Comment:
This can be `std::nullopt`, I think you need to do something like:

```suggestion
             << ", delete_retention_ms:"
             << c.delete_retention()
                  .value_or(std::chrono::milliseconds(-1))
                  .count()
```

Reply:
fixed